bee-runtime/src/worker.rs

```diff
@@ -18,3 +24,20 @@ impl std::error::Error for Error {
         self.0.source()
     }
 }
+
+#[async_trait]
+pub trait Worker<N: Node>: Any + Send + Sync + Sized {
```

Comment:
Not that important, but we basically do have our own actor framework here right? So should we maybe call it `Actor` instead of `Worker`? https://riker.rs or Actix do call it `Actor` too.
If we use the same terminologies we can easier compare and discuss with the other projects I guess, which might be helpful in the future.

Reply:
Good question, Sam. I think, however, that the semantics for `Actor`s are bit different than what we have implemented here. I see `Worker`s more like `Task`s or even `Thread`s rather than `Actor`s, because the actor model is also  tied to the concept of a `Supervisor` (respawning actors, ...) and an inbox/outbox (message) system.

Reply:
I also feel like it deviate a little bit from what we expect from an actor as our worker can represent a blocking work at start/stop or spawning tasks as well as registering/removing resources from a common context. I don't have a strong opinion but I think it's fine if we keep Worker. @zesterer wdyt?

Reply:
It is effectively an actor model, but I'm not convinced that this is reason enough to change the name by itself.

Reply:
I think actors are workers, but taken a step further. Actors share similarities with a microservice due to the stricter boundaries.

Reply:
Yes agree, we don't have a supervision strategy. But in the end I guess we need something like that too

Reply:
Right, that's a good point. We have a notion of node resources, i.e: shared state. Actor programming doesn't usually have that.

Reply:
Well, the node itself kinda is the supervisor tbh

Reply:
@thibault-martinez Yes, true. But we don't have the concept of respawning so i guess it's not fully fault-tolerant. 
reffering to something like this https://github.com/bastion-rs/bastion#guarantees

Reply:
Okey, so i guess we agreed on keeping the term "Worker" as it is :d

Reply:
yes, without a way to recover the node from a crashed worker I think we really cannot call the node a supervisor.